diff --git a/RedPandaIDE/debugger/debugger.cpp b/RedPandaIDE/debugger/debugger.cpp
index e4954044..dc56efdc 100644
--- a/RedPandaIDE/debugger/debugger.cpp
+++ b/RedPandaIDE/debugger/debugger.cpp
@@ -157,7 +157,8 @@ bool Debugger::startClient(int compilerSetIndex,
             if (mExecuting) {
                 stop();
             }
-            mTarget->deleteLater();
+            if (mTarget)
+                mTarget->deleteLater();
             mTarget = nullptr;
         });
         mTarget->addBinDirs(binDirs);
