diff --git a/libs/qsynedit/qsynedit/qsynedit.h b/libs/qsynedit/qsynedit/qsynedit.h
index 3b9ca9ce..76cbecb2 100644
--- a/libs/qsynedit/qsynedit/qsynedit.h
+++ b/libs/qsynedit/qsynedit/qsynedit.h
@@ -23,6 +23,7 @@
 #include <QFrame>
 #include <QStringList>
 #include <QTimer>
+#include <QElapsedTimer>
 #include <QWidget>
 #include "gutter.h"
 #include "codefolding.h"
diff --git a/RedPandaIDE/widgets/darkfusionstyle.cpp b/RedPandaIDE/widgets/darkfusionstyle.cpp
index d6028ad1..33075984 100644
--- a/RedPandaIDE/widgets/darkfusionstyle.cpp
+++ b/RedPandaIDE/widgets/darkfusionstyle.cpp
@@ -74,8 +74,8 @@ static QString uniqueName(const QString &key, const QStyleOption *option, const
     const QStyleOptionComplex *complexOption = qstyleoption_cast<const QStyleOptionComplex *>(option);
     QString tmp = QString("%1%2%3%4%5%6%7")
             .arg(key)
-            .arg(option->state)
-            .arg(option->direction)
+            .arg(uint(option->state))
+            .arg(int(option->direction))
             .arg(complexOption ? uint(complexOption->activeSubControls) : 0u)
             .arg(option->palette.cacheKey())
             .arg(size.width())
@@ -84,8 +84,8 @@ static QString uniqueName(const QString &key, const QStyleOption *option, const
 #if QT_CONFIG(spinbox)
     if (const QStyleOptionSpinBox *spinBox = qstyleoption_cast<const QStyleOptionSpinBox *>(option)) {
         tmp = tmp + QString("%1%2%3")
-                .arg(spinBox->buttonSymbols)
-                .arg(spinBox->stepEnabled)
+                .arg(int(spinBox->buttonSymbols))
+                .arg(uint(spinBox->stepEnabled))
                 .arg(QLatin1Char(spinBox->frame ? '1' : '0'));
     }
 #endif // QT_CONFIG(spinbox)
